api/coreservice.go

```diff
@@ -1589,6 +1591,11 @@ func (core *coreService) ActionsInActPool(actHashes []string) ([]action.SealedEn
 	return ret, nil
 }
 
+// Genesis returns the genesis of the chain
+func (core *coreService) Genesis() genesis.Genesis {
+	return core.bc.Genesis()
```

Comment:
coreService is the adapter for other services, where Genesis() shouldn't be exposed and tx decoding should be done

Reply:
there are some grpc-only interfaces in coreservice as well

Reply:
ok, that is not the reason to add another one ... here are my 3 thoughts when adding this to coreService:
1. there are existing similar funcs like `ChainID()`, `EvmNetworkID()`, `TipHeight()`, these actually all belong to the `Blockchain()` interface, so I don't see why cannot add a new `Genesis()`. We should have another PR to add a  `Blockchain()` funcs to deprecate these
2. like you said, we can add a `DecodeTx()` funcs instead of `Genesis()`, but given the facts in 1, doing so does not make our programming better IMO. The `DecodeTx()` is just a wrapper of some calls, it's not simpler or better readable than `Genesis()`
3. finally, after the hard-fork, we will switch to use the new code path, this `Genesis()` func won't be needed anymore